app/code/Magento/Catalog/Model/Product/Gallery/ImagesConfigFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ *
+ * Copyright © 2015 Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Catalog\Model\Product\Gallery;
+
+use \Magento\Framework\Data\CollectionFactory;
+use Magento\Framework\DataObject;
+
+class ImagesConfigFactory implements ImagesConfigFactoryInterface
+{
+    /**
+     * @var \Magento\Framework\Data\CollectionFactory
+     */
+    protected $_dataCollectionFactory;
```

Comment:
We do not encourage inheritance-based API, please make this property private. Additionally, naming convention with underscores is considered legacy, please rename this property.